File: backend/src/main/java/com/example/stamp_app/repository/AccountRepository.java
package com.example.stamp_app.repository;

import com.example.stamp_app.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;

import java.math.BigInteger;
import java.util.UUID;

public interface AccountRepository extends JpaRepository<Account, UUID> {
    Account findByEmail(String email);

    Account findByUuid(UUID uuid);

    Account findById(BigInteger id);
}

File: backend/src/main/java/com/example/stamp_app/repository/AccountRepository.kt
package com.example.stamp_app.repository

import com.example.stamp_app.entity.Account
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface AccountRepository : JpaRepository<Account?, UUID?> {

    fun findByEmail(email: String): Account?
    fun findByEmailAndDeletedAtIsNull(email: String): Account?
    fun findByUuid(uuid: UUID): Account?
    fun findById(id: Long): Account?

}
